package/MDAnalysis/core/_get_readers.py

```diff
@@ -202,7 +205,7 @@ def get_writer_for(filename, format=None, multiframe=None):
         raise TypeError(errmsg.format(format)) from None
 
 
-def get_parser_for(filename, format=None):
+def get_parser_for(filename: Union[str, mmtf.MMTFDecoder], format: Optional[str] = None) -> _PARSERS:
```

Comment:
This method is an example of a method where the docstring is out of date. I expect that you will encounter many of those. Fixing them would be of great benefit.

In this method, `filename` can be anything. Like in `get_reader_for`, the `filename` argument will be passed to a series of functions to figure out what parser is appropriate. The docstring does not reflect that and should.

The `format` argument can either be a string, in which case it would be the name of a format, or a class describing a parser.